File: game_server/ClientProtocol/GC2GS/p021_PlayerInfo/GC2GS_021_046_ReqClientNotifyFuncUnlock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GC2GS.p021_PlayerInfo
{

public class GC2GS_021_046_ReqClientNotifyFuncUnlock : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 功能类型
/// </summary>
private NPEnum.ENPFunctionType funcType;


public GC2GS_021_046_ReqClientNotifyFuncUnlock() {
	funcType = 0;
}

public GC2GS_021_046_ReqClientNotifyFuncUnlock(
	NPEnum.ENPFunctionType _funcType
) {	funcType = _funcType;
}

public byte getMainOrder() { return (byte)21; }

public byte getSubOrder() { return (byte)46; }

/// <summary>
/// 功能类型
/// </summary>
public NPEnum.ENPFunctionType getFuncType() { return funcType; }
/// <summary>
/// 功能类型
/// </summary>
public void setFuncType(NPEnum.ENPFunctionType _funcType) { funcType = _funcType; }


public int GetBufSize() {
	int _size = 4;

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 6;

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	funcType = (NPEnum.ENPFunctionType)_buf.getInt();
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putInt((int)funcType);

}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)21);
	_buf.put((byte)46);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)21);
	_recBuf.put((byte)46);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("funcType").Append(":").Append(funcType.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

File: game_server/ServerProtocol/src/GC2GS/p021_PlayerInfo/GC2GS_021_046_ReqClientNotifyFuncUnlock.java
package GC2GS.p021_PlayerInfo;

import java.nio.ByteBuffer;
public class GC2GS_021_046_ReqClientNotifyFuncUnlock implements ALBasicProtocolPack._IALProtocolStructure {
/** 功能类型 */
private NPEnum.ENPFunctionType funcType;


public GC2GS_021_046_ReqClientNotifyFuncUnlock() {
	funcType = NPEnum.ENPFunctionType.values()[0];
}

public GC2GS_021_046_ReqClientNotifyFuncUnlock(
	 NPEnum.ENPFunctionType _funcType
) {	funcType = _funcType;
}

public final byte getMainOrder() { return (byte)21; }

public final byte getSubOrder() { return (byte)46; }

/** 功能类型 */
public NPEnum.ENPFunctionType getFuncType() { return funcType; }
/** 功能类型 */
public void setFuncType(NPEnum.ENPFunctionType _funcType) { funcType = _funcType; }


public final int GetBufSize() {
	int _size = 4;

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 6;

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) funcType = NPEnum.ENPFunctionType.ENPFunctionType_FromInt(_buf.getInt());
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putInt(funcType.ordinal());

}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)21);
	_buf.put((byte)46);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)21);
	_recBuf.put((byte)46);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

